ShortestPathLength/ShortestPathLength/Program.cs

```diff
@@ -23,33 +23,32 @@ public static int shortestPathLength(Node start, Node end)
 
             // Start at the start
             queue.Enqueue(start);
-            scratch.Add(start, start.id);
+            scratch.Add(start, distance);
 
             // Create a set of nodes S that are reachable from start
             while (queue.Count != 0)
             {
+                distance += 1;
+                
                 Node n = queue.Dequeue();
 
                 foreach (Node nei in n.neighbors)
                 {
                     if (!scratch.ContainsKey(nei))
                     {
                         queue.Enqueue(nei);
-                        scratch.Add(nei, nei.id);
-
+                        scratch.Add(nei, distance);
```

Comment:
You have to compute distance.  The single variable isn't enough to keep track of what ring you're in.